### PR TITLE
HADotNet - using IHttpClientFactory

### PR DESCRIPTION
**Pull request by imalkevich**
Hi @qJake , @Devqon,

I made another attempt after taking inspiration from the following articles:
* https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
* https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0#consumption-patterns
* https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/implement-http-call-retries-exponential-backoff-polly

So in short I think we should use IHttpClientFactory as suggested by Microsoft developers. In this case we can add retry mechanism easily using [Polly library](https://github.com/App-vNext/Polly).

I tried to polish all the changes so this should be ready to merged, but please let me know if you find somethings.

Thanks

### CODE REVIEW

**Comment by imalkevich** [comment]
Hi @qJake , @Devqon,

I made another attempt after taking inspiration from the following articles:
* https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
* https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0#consumption-patterns
* https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/implement-http-call-retries-exponential-backoff-polly

So in short I think we should use IHttpClientFactory as suggested by Microsoft developers. In this case we can add retry mechanism easily using [Polly library](https://github.com/App-vNext/Polly).

I tried to polish all the changes so this should be ready to merged, but please let me know if you find somethings.

Thanks

**Review by bogdanbujdea — APPROVED** [review]

**Comment by imalkevich** [comment]
Hi Jake @qJake ,

Please review/merge this once you have time.

Thanks
